### DISCUSSION TITLE
Extracting tenant ID from Prometheus labels

### DISCUSSION THREAD

**Discussion by jmajoor** [grafana/mimir#2004] [Help and support]
In the intro webinar one use case for mimir is separation between multiple teams in a single cluster with a single prometheus server.

The only thing I could find on how to really implement that is by using cortex-tenant, but then I have to use yet another component and this one is not part of grafana labs.

How do I configure Mimir to isolate different teams in a single cluster?
If cortex-tenant is the only way, then are there any plans to properly support this scenario out-of-the-box?

**Comment by pstibrany** [comment]
> In the intro webinar one use case for mimir is separation between multiple teams in a single cluster with a single prometheus server.

Did we promote this use case using single Prometheus server? Normally we would advocate for separate Prometheus server per team, and writing to separate tenants.

Alternative would be to use Label-Based Access Control feature available in Grafana Enterprise Metrics.

**Reply by jmajoor** [reply]
The use case showed an architecture diagram with multiple teams in a single cluster with a single prometheus server, but the use case was not explained in detail during the webinar. The diagram immediately caught my eye, as that would be ideal for what we want to achieve. The overhead of having a prometheus server per team is quite significant (in our scenario).

I would prefer label based tenancy over label based access control, is this something Mimir would consider?
For example instead of using the http header to determine the tenant on ingress, to be able to configure a label to determine the tenant upon ingress.

**Reply by pracucci** [reply]
> I would prefer label based tenancy over label based access control, is this something Mimir would consider?

Currently, label based access control is a feature of Grafana Enterprise Metrics (GEM), which is a Grafana Labs licensed product based on Grafana Mimir:
https://grafana.com/docs/enterprise-metrics/latest/tenant-management/lbac/

**Reply by pracucci** [reply]
> The only thing I could find on how to really implement that is by using cortex-tenant, but then I have to use yet another component and this one is not part of grafana labs.

cortex-tenant (which we also [reference in our doc](https://grafana.com/docs/mimir/next/operators-guide/securing/authentication-and-authorization/#extracting-tenant-id-from-prometheus-labels)) is the only OSS tool I'm aware of right now, and it's what I would suggest to begin with.
